File: android/app/src/main/kotlin/com/menumaker/data/remote/models/AuthModels.kt
package com.menumaker.data.remote.models

import com.google.gson.annotations.SerializedName

// Request models
data class LoginRequest(
    @SerializedName("email") val email: String,
    @SerializedName("password") val password: String
)

data class SignupRequest(
    @SerializedName("email") val email: String,
    @SerializedName("password") val password: String,
    @SerializedName("name") val name: String,
    @SerializedName("phone") val phone: String? = null
)

data class RefreshTokenRequest(
    @SerializedName("refresh_token") val refreshToken: String
)

// Response models
data class AuthResponse(
    @SerializedName("success") val success: Boolean,
    @SerializedName("data") val data: AuthData
)

data class AuthData(
    @SerializedName("access_token") val accessToken: String,
    @SerializedName("refresh_token") val refreshToken: String,
    @SerializedName("user") val user: UserDto
)

data class UserDto(
    @SerializedName("id") val id: String,
    @SerializedName("email") val email: String,
    @SerializedName("name") val name: String,
    @SerializedName("phone") val phone: String?,
    @SerializedName("role") val role: String,
    @SerializedName("created_at") val createdAt: String
)

File: ios/MenuMaker/Data/Models/AuthModels.swift
import Foundation

// MARK: - Request Models

struct LoginRequest: Encodable {
    let email: String
    let password: String
}

struct SignupRequest: Encodable {
    let email: String
    let password: String
    let name: String
    let phone: String?
}

struct RefreshTokenRequest: Encodable {
    let refreshToken: String
}

struct ForgotPasswordRequest: Encodable {
    let email: String
}

struct UpdateProfileRequest: Encodable {
    let name: String?
    let phone: String?
    let address: String?
}

struct ChangePasswordRequest: Encodable {
    let currentPassword: String
    let newPassword: String
}

// MARK: - Response Models

struct AuthResponse: Decodable {
    let success: Bool
    let data: AuthData
}

struct AuthData: Decodable {
    let accessToken: String
    let refreshToken: String
    let user: User
}

struct UserResponse: Decodable {
    let success: Bool
    let data: UserData
}

struct UserData: Decodable {
    let user: User
}

struct MessageResponse: Decodable {
    let success: Bool
    let message: String
}

struct User: Codable, Identifiable {
    let id: String
    let email: String
    let name: String
    let phone: String?
    let address: String?
    let photoUrl: String?
    let role: String
    let businessId: String?  // Business ID for seller accounts
    let createdAt: String
    let updatedAt: String?

    var isAdmin: Bool {
        role == "admin"
    }

    var isSeller: Bool {
        role == "seller" || role == "admin"
    }

    var formattedCreatedDate: String {
        guard let date = ISO8601DateFormatter().date(from: createdAt) else {
            return createdAt
        }

        let formatter = DateFormatter()
        formatter.dateStyle = .medium
        return formatter.string(from: date)
    }

    var formattedPhone: String {
        phone ?? "Not provided"
    }

    var formattedAddress: String {
        address ?? "Not provided"
    }
}
